### PR TITLE
[DOC-107] add a page for Data types in Explore YCQL features

### PR DESCRIPTION
**Pull request by aravind-nallan-yb**
New content for the Data types page on Explore YCQL features section

@netlify preview/explore/ycql-language/data-types/

### CODE REVIEW

File: docs/content/stable/explore/ycql-language/data-types.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
+---
+title: Data types
+linkTitle: Data types
+description: Data types in YCQL
+image: /images/section_icons/secure/create-roles.png
+menu:
+  preview:
```
Comment (ddhodge) [inline docs/content/stable/explore/ycql-language/data-types.md:7]:
```suggestion
  stable:
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 2]
```diff
@@ -0,0 +1,470 @@
+---
+title: Data types
+linkTitle: Data types
+description: Data types in YSQL
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:4]: Looks like the YSQL page, not YCQL?

Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:4]:
```suggestion
description: Data types in YCQL
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+image: /images/section_icons/secure/create-roles.png
+menu:
+  preview:
+    identifier: explore-ysql-language-features-data-types
+    parent: explore-ysql-language-features
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:9]:
```suggestion
    parent: explore-ycql-language
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 4]
```diff
@@ -0,0 +1,470 @@
+---
+title: Data types
+linkTitle: Data types
+description: Data types in YSQL
+image: /images/section_icons/secure/create-roles.png
+menu:
+  preview:
+    identifier: explore-ysql-language-features-data-types
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:8]:
```suggestion
    identifier: explore-ycql-language-features-data-types
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+    parent: explore-ysql-language-features
+    weight: 150
+type: docs
+---
+
+This document describes the data types supported in YSQL, from the basic data types to the `SERIAL` pseudo-type (for implementing an auto-incrementing primary key column in a table), arrays, composite types, and range types.
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:14]: wrong content?

**Review by ddhodge — COMMENTED** [review]
Fix frontmatter

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+```
+
+```output
+ user_id                              | user_name | recent_logins                                                          | phone_numbers                                    | account_numbers
+--------------------------------------+-----------+------------------------------------------------------------------------+--------------------------------------------------+-----------------
+ 28df63b7-cc57-43cb-9752-fae69d1653da |  John Doe | ['2011-02-03 04:05:00.000000+0000', '2023-04-05 09:15:08.000000+0000'] | {'home': '669-555-1212', 'work': '408-555-2121'} |  {'sa-1011212'}
+
+(1 rows)
+```
+The above example appends the new element to an existing list. Prepending is also possible as shown
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:241]:
```suggestion
The preceding example appends the new element to an existing list. Prepending is also possible, as follows:
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+below:
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:242]:
```suggestion
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+
+```output
+ user_id                              | user_name | recent_logins                                                          | phone_numbers                                    | account_numbers
+--------------------------------------+-----------+------------------------------------------------------------------------+--------------------------------------------------+-----------------
+ 28df63b7-cc57-43cb-9752-fae69d1653da |  John Doe | ['2023-04-05 09:15:08.000000+0000', '2011-02-03 04:05:00.000000+0000'] | {'home': '669-555-1212', 'work': '408-555-2121'} |  {'sa-1011212'}
+
+(1 rows)
+```
+
+Working with `SET` and `MAP` is similar to the above example except the these types do not have a
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:262]:
```suggestion
`SET` and `MAP` work similarly, except the these types do not have a
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+notion of prepending and the syntax for literals is slightly different. See [YCQL
+Collections](../../../api/ycql/type_collection) for more details.
+
+## User defined types
+
+A user defined type is a collection of data types similar to a `struct` in a programming language.
+
+### 1. Create a composite type
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:270]: Steps shouldn't be separate headings.

Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:270]:
```suggestion
1. Create a composite type as follows:
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+
+```sql
+CREATE TYPE inventory_item (
+   name text,
+   supplier_id integer,
+   price float
+);
+```
+
+### 2. Create table with a composite type
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:280]:
```suggestion
1. Create a table with a composite type as follows:
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+CREATE TABLE on_hand (
+   item_id UUID PRIMARY KEY,
+   item inventory_item,
+   count integer
+);
+```
+
+### 3. Insert a row
+
+To insert a row as follows:
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:292]:
```suggestion
1. Insert a row as follows:
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+
+```sql
+CREATE TABLE on_hand (
+   item_id UUID PRIMARY KEY,
+   item inventory_item,
+   count integer
+);
+```
+
+### 3. Insert a row
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:290]:
```suggestion
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+
+To insert a row as follows:
+
+```sql
+INSERT INTO on_hand (item_id, item, count) VALUES (28df63b7-cc57-43cb-9752-fae69d1653da, {name: 'fuzzy dice', supplier_id: 42, price: 1.99}, 1000);
+```
+
+### 4. Select data
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:298]:
```suggestion
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+
+To select data from the `on_hand` example table, execute the following:
```
Comment (ddhodge) [inline docs/content/preview/explore/ycql-language/data-types.md:300]:
```suggestion
1. To select data from the `on_hand` example table, execute the following:
```

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+
+`varchar` and `text` are aliases.
+
+The following Apache Cassandra character types are not supported:
+
+* `ascii`: Use `text` or `varchar`
+
+To test YugabyteDB support for character types, create a table that has columns with the following types specified:
+
+```sql
```
[the remark this answers is not in the document]
Comment (aravind-nallan-yb) [reply docs/content/preview/explore/ycql-language/data-types.md:35]: Good catch. My bad. I was use an existing keyspace.

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+CREATE TABLE temporal_types (
+  date_type DATE PRIMARY KEY,
+  time_type TIME,
+  timestamp_type TIMESTAMP,
+);
+```
+
+The following example inserts a row into the table:
+
+```sql
```
[the remark this answers is not in the document]
Comment (aravind-nallan-yb) [reply docs/content/preview/explore/ycql-language/data-types.md:120]: Looks like i didn't test these very well. Fixed now.

File: docs/content/preview/explore/ycql-language/data-types.md [hunk 17]
```diff
@@ -0,0 +1,328 @@
...
+notion of prepending and the syntax for literals is slightly different. See [YCQL
+Collections](../../../api/ycql/type_collection) for more details.
+
+## User defined types
+
+A user defined type is a collection of data types similar to a `struct` in a programming language.
+
+1. Create a user defined type
+
+```sql
```
[the remark this answers is not in the document]
Comment (aravind-nallan-yb) [reply docs/content/preview/explore/ycql-language/data-types.md:288]: Ah! i was wondering why you had all 1s in ur suggested changes earlier.

**Review by ddhodge — APPROVED** [review]

**Pull request merged**